### PR TITLE
Add initial_state option to mpl drawer

### LINKED ISSUE #4293: initial_state=True is not honored by mpl drawer
When mpl drawer is called with `initial_state=True` the initial values are not there:
```
from qiskit import *
qr = QuantumRegister(3, 'q')
cr = ClassicalRegister(3, 'c')
circuit = QuantumCircuit(qr, cr)
circuit.measure(qr, cr)
circuit.draw('mpl', initial_state=True)
```
image

The output should be something like this:

circuit

### PR DESCRIPTION
**Pull request by enavarro51**
### Summary

Fix #4293

### Details and comments

Added user option in mpl drawer for initial_state bool to display initial state next to qubits and clbits.

### CODE REVIEW

File: qiskit/visualization/circuit_visualization.py [hunk 1]
```diff
@@ -137,7 +137,7 @@ def circuit_drawer(circuit,
             when the ``output`` kwarg is set to use the ``mpl`` backend. It
             will be silently ignored with all other outputs.
         initial_state (bool): Optional. Adds ``|0>`` in the beginning of the wire.
-            Only used by the ``text``, ``latex`` and ``latex_source`` outputs.
+            Used by the ``text``, 'mpl', ``latex`` and ``latex_source`` outputs.
```
Comment (1ucian0) [inline qiskit/visualization/circuit_visualization.py:140]: If it's for all of them, we can remove that sentence (as in other for-all args)
```suggestion
```

File: qiskit/visualization/circuit_visualization.py [hunk 2]
```diff
@@ -635,6 +637,7 @@ def _matplotlib_circuit_drawer(circuit,
         idle_wires (bool): Include idle wires. Default is True.
         with_layout (bool): Include layout information, with labels on the physical
             layout. Default: True.
+        initial_state (bool): Optional. Adds |0> in the beginning of the line. Default: `True`.
```
Comment (1ucian0) [inline qiskit/visualization/circuit_visualization.py:640]:
```suggestion
        initial_state (bool): Optional. Adds ``|0>`` in the beginning of the wire. Default: `True`.
```

File: qiskit/visualization/circuit_visualization.py [hunk 3]
```diff
@@ -615,6 +616,7 @@ def _matplotlib_circuit_drawer(circuit,
                                justify=None,
                                idle_wires=True,
                                with_layout=True,
+                               initial_state=True,
```
Comment (1ucian0) [inline qiskit/visualization/circuit_visualization.py:619]:
```suggestion
                               initial_state=False,
```

Sorry for the misunderstanding in https://github.com/Qiskit/qiskit-terra/issues/4293#issuecomment-622654114. By consistency I mean with `draw`. MPL is not been tested now.

File: qiskit/visualization/circuit_visualization.py (hunk 1 above)
Comment (enavarro51) [reply qiskit/visualization/circuit_visualization.py:140]: Had the same thought.

File: qiskit/visualization/circuit_visualization.py (hunk 2 above)
Comment (enavarro51) [reply qiskit/visualization/circuit_visualization.py:640]: Also the default should be 'False'?

File: qiskit/visualization/circuit_visualization.py (hunk 3 above)
Comment (enavarro51) [reply qiskit/visualization/circuit_visualization.py:619]: Done. Also set it to False for matplotlib.py in the __init__ in MatplotlibDrawer.

**Comment by enavarro51** [comment]
I'll make the changes on my local and push the commit. I haven't done a reno before. Can you walk me through the steps? Thanks.

File: qiskit/visualization/circuit_visualization.py (hunk 2 above)
Comment (1ucian0) [reply qiskit/visualization/circuit_visualization.py]: yeap.

**Comment by 1ucian0** [comment]
> Can you walk me through the steps?

It's explained here, https://github.com/Qiskit/qiskit-terra/blob/master/CONTRIBUTING.md#release-notes. Let me know if something there is not clear (making documentation clearer is important, we can fix it later). The tl;dr is:
Run `reno new initial_state_all_drawers`. In the created file, remove everything but the `features` section. Explain there that the parameter now applies to all the drawer and maybe add an example of how it works for mpl. Check the results with `reno report`.

**Comment by mtreinish** [comment]
From a quick glance this lgtm now, can you resolve the merge conflicts and then I'll re-review in depth

**Comment by enavarro51** [comment]
@mtreinish I resolved the merge conflict. I did all the changes suggested by @1ucian0 previously on my local and pushed them here.

**Comment by 1ucian0** [comment]
MPL circuit drawer issues are piling up, since we dont have a good way to track changes on it. I'm putting this PR `on hold` (meaning "do not merge yet") until https://github.com/Qiskit/qiskit-terra/pull/4544 (or a similar solution) gets in.

**Comment by 1ucian0** [comment]
#4544 is merged... Let's add a test with the new method!

I will try helping, but fill free to do it yourself and ask me if any question.

**Comment by 1ucian0** [comment]
This one was covered by PR #4616.

**Pull request closed without merging**